File: go/insertionSort.go
package main

func insertionSort(A []int) []int {
	i := 1
	for i < len(A) {
		j := i
		for j > 0 && A[j-1] > A[j] {
			A[j], A[j-1] = A[j-1], A[j]
			j = j - 1
		}
		i = i + 1
	}

	return A
}

File: javascript/insertionSort.js
function InsertionSort(A)
{
    let i = 1
    while(i < A.length)
    {
        let j = i
        while(j > 0 && A[j-1] > A[j])
        {
            let temp = A[j]
            A[j] = A[j-1]
            A[j-1] = temp
            j -= 1
        }
        i += 1
    }

    return A
}
